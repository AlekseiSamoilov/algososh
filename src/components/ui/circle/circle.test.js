import { render } from "@testing-library/react";
import { Circle } from "./circle"
import { ElementStates } from "../../../types/element-states";


describe('Circle component', () => {
    test('renders correctly without letter', () => {
        const { asFragment } = render(<Circle />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly with letter', () => {
        const { asFragment } = render(<Circle letter={"A"} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly with head', () => {
        const { asFragment } = render(<Circle head={'A'} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly with react-element in head', () => {
        const reactElement = <div>Element</div>
        const { asFragment } = render(<Circle head={reactElement} />);
        expect(asFragment()).toMatchSnapshot();
    });
    
    test('renders correctly with tail', () => {
        const { asFragment } = render(<Circle tail={'A'} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly with react-element in tail', () => {
        const reactElement = <div>Element</div>
        const { asFragment } = render(<Circle tail={reactElement} />);
        expect(asFragment()).toMatchSnapshot();
    });
    
    test('renders correctly with index', () => {
        const { asFragment } = render(<Circle index={4} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly with isSmall prop', () => {
        const { asFragment } = render(<Circle isSmall={true} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly whith Default state', () => {
        const { asFragment } = render(<Circle state={ElementStates.Default} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly whith Changing state', () => {
        const { asFragment } = render(<Circle state={ElementStates.Changing} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders correctly whith Modified state', () => {
        const { asFragment } = render(<Circle state={ElementStates.Modified} />);
        expect(asFragment()).toMatchSnapshot();
    });
})