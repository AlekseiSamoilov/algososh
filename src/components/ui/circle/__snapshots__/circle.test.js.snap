// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component renders correctly whith Changing state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   changing"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly whith Default state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly whith Modified state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   modified"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
      data-testid="circle-head"
    >
      A
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      4
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with isSmall prop 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle  small default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with react-element in head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    >
      <div>
        Element
      </div>
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with react-element in tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    >
      <div>
        Element
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly with tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
      data-testid="circle-tail"
    >
      A
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle component renders correctly without letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;
